vault/capabilities.go

```diff
@@ -0,0 +1,83 @@
+package vault
+
+import (
+	"fmt"
+	"sort"
+	"strings"
+)
+
+// CapabilitiesResponse holds the result of fetching the capabilities of token on a path
+type CapabilitiesResponse struct {
+	Root         bool
+	Capabilities []string
+}
+
+// Capabilities is used to fetch the capabilities of the given token on the given path
+func (c *Core) Capabilities(token, path string) (*CapabilitiesResponse, error) {
+	if path == "" {
+		return nil, fmt.Errorf("missing path")
+	}
+
+	if token == "" {
+		return nil, fmt.Errorf("missing token")
+	}
+
+	te, err := c.tokenStore.Lookup(token)
+	if err != nil {
+		return nil, err
+	}
+	if te == nil {
+		return nil, fmt.Errorf("invalid token")
+	}
+
+	if te.Policies == nil {
+		return nil, nil
+	}
+
+	var result CapabilitiesResponse
+	capabilities := make(map[string]bool)
+	for _, tePolicy := range te.Policies {
+		if tePolicy == "root" {
+			result.Root = true
+			break
+		}
+		policy, err := c.policyStore.GetPolicy(tePolicy)
+		if err != nil {
+			return nil, err
+		}
+		if policy == nil || policy.Paths == nil {
+			continue
+		}
+		for _, pathCapability := range policy.Paths {
+			switch pathCapability.Glob {
+			case true:
```

Comment:
While this probably works correctly it easily can look odd to have "case true". I'd simply use
`switch {
case pathCapability.Glob:
default:
}`